### DISCUSSION TITLE
Hardware SPI implementation in C for SAM V70

### DISCUSSION THREAD

**Discussion by ddlqn** [olikraus/u8g2#1421] [Q&A]
Hi all,

I was wondering whether anyone has replaced one of the software SPI functions (e.g. u8x8_byte_3wire_sw_spi) with a hardware implementation and/or whether there is any documentation, that I have not uncovered yet, describing how to go about it.

The refresh of the displays I am using is fairly sluggish because I need to run other tasks in parallel, so as a first step I am looking into using the hardware SPI interface of the controller (SAM V70).

Thanks,
Daniel

**Comment by v1993** [comment]
Wiki provides a fairly good overview on how to do I/O, including HW SPI (see [this section](https://github.com/olikraus/u8g2/wiki/Porting-to-new-MCU-platform#communication-callback-eg-u8x8_byte_hw_i2c)). It varies wildly from uC to uC, so I recommend adding "for SAM V70" to question title.

**Reply by ddlqn** [reply]
Thanks, I thought I had seen it before but for some reason was unable to find it.
